### PR TITLE
Add leader election

### PR DESCRIPTION
**Pull request by WesleyAC**
This is untested, but I think that all the features are here.

Relies on #8.

### CODE REVIEW

File: src/node.py [hunk 1]
```diff
@@ -33,10 +49,44 @@ def update_term(self, term):
...
+        if type(message) == AppendEntries:
+            if self.node_type == "F":
+                self.broker.add_timer(self.node_id, self.election_timeout)
+            elif self.node_type == "C" or self.node_type == "L":
+                self.change_type("F")
+        elif type(message) == RequestVote:
+            if message.term < self.term or self.voted_for is not None:
+                self.broker.send_to(self.node_id, sender, RequstVoteResponse(self.term, False))
+            else:
+                self.broker.send_to(self.node_id, sender, RequstVoteResponse(self.term, True))
```
Comment (ghost) [inline src/node.py:71]: Shouldn't we also check that candidate's log is at least as up-to-date as receiver's log?

File: src/node.py [hunk 2]
```diff
@@ -33,10 +49,44 @@ def update_term(self, term):
...
+        elif type(message) == AppendEntriesResponse:
+            pass
```
Comment (ghost) [inline src/node.py:73]: We are going to fill this up right?

File: src/node.py (hunk 1 above)
Comment (WesleyAC) [reply src/node.py:71]: Yes, we should :)

Thanks!

File: src/node.py (hunk 2 above)
Comment (WesleyAC) [reply src/node.py:73]: Yeah. It's not important for leader election though, so I left it out.

**Pull request merged**